### ISSUE TITLE
dot style assignment not works with indexed object interface

### ISSUE THREAD

**Issue by e-cloud** [Microsoft/TypeScript#12611]
**TypeScript Version:**  2.1.1

**Code**

```ts
interface Dictionary {
    [key: string]: any;
}

const plugins: Dictionary = {}

plugins.test = {}

const plugins2: {
    [key: string]: any;
} = {}

plugins2.test = {}
```

**Expected behavior:**
no error

**Actual behavior:**
```
Property 'test' does not exist on type 'Dictionary'.
any
```

**[See the playground example](https://www.typescriptlang.org/play/index.html#src=interface%20Dictionary%20%7B%0D%0A%20%20%20%20%5Bkey%3A%20string%5D%3A%20any%3B%0D%0A%7D%0D%0A%0D%0Aconst%20plugins%3A%20Dictionary%20%3D%20%7B%7D%0D%0A%0D%0Aplugins.test%20%3D%20%7B%7D%0D%0A%0D%0Aconst%20plugins2%3A%20%7B%0D%0A%20%20%20%20%5Bkey%3A%20string%5D%3A%20any%3B%0D%0A%7D%20%3D%20%7B%7D%0D%0A%0D%0Aplugins2.test%20%3D%20%7B%7D)**

IMHO, dot style assignment should infer the property name as `string`

I think it's related to https://github.com/Microsoft/TypeScript/issues/6041

**Comment by mhegazy** [comment]
Duplicate of https://github.com/Microsoft/TypeScript/issues/12596

**Comment by e-cloud** [comment]
this has been fixed in v2.2
